inc/group_group.class.php

```diff
@@ -184,6 +184,11 @@ function getGroups($ticket_id, $removeAlreadyAssigned = true) {
          $groupname = $group_obj->fields['name'];
       }
 
+      //if empty new tickets return user groups
+      if (empty($groups) && CommonDBTM::isNewID($ticket_id)) {
+         return Group_User::getUserGroups($_SESSION['glpiID']);
```

Comment:
I don't think checking if the ticket is new is the correct way to handle this.
A ticket may be created without groups, or its groups may be deleted after a while.

In both this cases, it seems valid to allow the user to assign any of his groups to the ticket.
I think we should probably check that the tickets groups count is 0.

Reply:
the variable ```$groups``` already refers to the ticket's groups, but only if the ticket exists (```$id > 0```) see ```L144```